webview-ui/src/components/ChatView.tsx

```diff
@@ -29,30 +43,71 @@ export const ChatView = ({
   errorMessage,
 }: ChatViewProps) => {
   const [userPrompt, setUserPrompt] = useState("");
-
   const extensionMessenger = new ExtensionMessenger();
+  let innerTextArea: HTMLTextAreaElement | null | undefined = null;
+  const endOfMessagesRef = useRef<HTMLDivElement>(null);
 
-  const handleInputOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newval = event.target.value;
-    setUserPrompt(newval);
+  /**
+   * Scroll to the bottom of the chat (if messages are long enough to scroll).
+   */
+  const scrollToBottom = () => {
+    if (endOfMessagesRef.current) {
+      scrollIntoView(endOfMessagesRef.current, {
+        behavior: "smooth",
+        scrollMode: "if-needed",
+      });
+    }
   };
 
-  const handleSubmit = (
-    e:
-      | React.MouseEvent<HTMLButtonElement, MouseEvent>
-      | React.FormEvent<HTMLFormElement>
-      | React.KeyboardEvent<HTMLInputElement>,
-  ) => {
-    e.preventDefault();
+  /**
+   * Only handle sending the message if not Shift+Enter key combination.
+   */
+  const onKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (event.key === "Enter" && !event.shiftKey) {
+      event.preventDefault();
+      event.stopPropagation();
+      handleSubmit();
+    }
+  };
+
+  /**
+   * Update user prompt and resize text area if necessary and possible.
+   */
+  const onInput = (event: InputEvent) => {
+    const target = event.target as HTMLTextAreaElement;
+    setUserPrompt(target.value);
+
+    if (innerTextArea) {
+      autosize.update(innerTextArea);
+    } else {
+      innerTextArea = document
+        .getElementById("vscode-textarea-chat-input")
+        ?.shadowRoot?.querySelector("textarea");
+      if (innerTextArea) {
```

Comment:
`innerTextArea` should be the `<textarea>` in the shadow DOM of the `<VSCodeTextArea>` component. The only reason this wouldn't be found by the check on line 86 is if the [Webview UI Toolkit](https://github.com/microsoft/vscode-webview-ui-toolkit) changes their implementation of `<VSCodeTextArea>` to not even use a `<textarea>`. Unlikely, but technically possible.

I've added a console error message explaining that condition. If that is the case, then (at least with the current implementation of `<VSCodeTextArea>`, the input box will just be scrollable instead of auto-resizing. I have no idea what would happen if the Webview UI Toolkit implemented it differently. We'll just have to check for breaking changes on that component library. Is there a way you'd recommend noting that dependency?